torchtune/modules/loss/kd_losses.py

```diff
@@ -137,4 +137,7 @@ def forward(
                 student_chunk, teacher_chunk, label_chunk, normalize=False
             )
 
+        if torch.sum(mask.view(-1), dim=0) == 0:
+            return torch.tensor(0.0, device=student_logits[0].device)
```

Comment:
In the discussion on #2094, it seems like if we hit this point, there's probably something wrong with the training data. In that case, should we error out here? Or drop a warning? Or let people shoot themselves in the foot?

cc @lindawangg and @ebsmothers

Reply:
I will wait for inputs, and will update #2094 after this PR is resolved.

Reply:
Any inputs?
@lindawangg , @ebsmothers

Reply:
Thanks for the bump here. Personally I think this is the right way to handle it. Yes, we could error/raise a warning, but honestly this behavior should be pretty obvious in a loss curve as it will clearly just drop to zero on any such iterations (plus I think zero loss would technically be the "correct" loss value if every token is being ignored). No need to overdo it on handling this edge case here imo